### PR TITLE
Hl fixes

### PR DESCRIPTION
**Pull request by mnunberg**
Contains docs, and revised API as discussed

### CODE REVIEW

File: docs/Highlight.md [hunk 1]
```diff
@@ -6,115 +6,84 @@ It would be good to support multiple highlighter engines, but also allow for
...
-
-* **num** number of fields to summarize
-* **field** fields to summarize
-
-The return value will be augmented to the fields returned via `RETURN`.
-Conversely, if `RETURN` appears after this keyword, then the fields indicated
-in `RETURN` follow after those from `SUMMARIZE`.
+FT.SEARCH ...
+    SUMMARIZE [FIELDS {num} {field}] [FRAGS {numFrags}] [LEN {fragLen}]
+    HIGHLIGHT [FIELDS {num} {field}] [TAGS {openTag} {closeTag}]
```
Comment (dvirsky) [inline docs/Highlight.md]: what about separator?
also len should be the total len, not the frag len.

File: docs/Highlight.md [hunk 2]
```diff
@@ -6,115 +6,84 @@ It would be good to support multiple highlighter engines, but also allow for
...
+    specified, a built-in tag value is appended and prepended.
+
+
+#### Field Selection
+
+If no specific fields are passed to the `RETURN`, `SUMMARIZE`, or `HIGHLIGHT`
+keywords, then all of a document's fields are returned. However, if any of these
+keywords contain a `FIELD` directive, then the `SEARCH` command will only retun
+the sum total of all fields enumerated in any of those directives. For example
+in the command `RETURN 1 foo SUMMARIZE FIELDS 1 bar HIGHLIGHT FIELDS 1 baz`,
```
Comment (dvirsky) [inline docs/Highlight.md]: That is not the behavior we agreed on.
1. Return always wins. You can't contradict it and return fields that are not mentioned in it.
2. If HIGHLIGHT or SUMMARIZE fields are a superset of RETURN, we ignore anything not in RETURN.
3. highlighted fields and summarized fields can be disjointed but they must obey return.

File: src/pytest/test.py [hunk 3]
```diff
@@ -1433,7 +1437,7 @@ def testSummarizationMeta(self):
 
         # Now, return the fields:
         res = self.cmd('ft.search', 'idx', 'pill pillow piller',
-                       'RETURN', 1, 'baz', 'SUMMARIZE', 2, 'foo', 'bar')
+                       'RETURN', 1, 'baz', 'SUMMARIZE', 'FIELDS', 2, 'foo', 'bar')
```
Comment (dvirsky) [inline src/pytest/test.py]: in this case we should return just baz

File: docs/Highlight.md (hunk 1 above)
Comment (mnunberg) [reply docs/Highlight.md]: The total len of what? if so, why allow the number of fragments? `LEN` indicates how much context each fragment should have, and `FRAGS` indicates the maximum number of samples of the matching term should be returned.

`LEN` cannot be used to limit the size of the block returned. In theory it's possible for a single fragment to be very very large; if a matching term exists repeatedly in the same field. This is actually something we should fix...

**Comment by mnunberg** [comment]
@dvirsky I believe all addressed issues should be resolved by now.

**Comment by dvirsky** [comment]
Thanks. I'll merge soon.

**Pull request merged**